File: pdu/python/std_msgs/pdu_cdr_conv_Float32.py
from .pdu_pytype_Float32 import Float32
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Float32(writer: CdrWriter, src: Float32):
    writer.write_float32(src.data)


def cdr_body_to_py_Float32(reader: CdrReader, dst: Float32):
    dst.data = reader.read_float32()
    return dst


def py_to_cdr_Float32(src: Float32) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Float32(writer, src)
    return writer.bytes()


def cdr_to_py_Float32(cdr_payload) -> Float32:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Float32()
    return cdr_body_to_py_Float32(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Float32.js
import { Float32 } from './pdu_jstype_Float32.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduFloat32Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Float32 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_float32(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Float32 } dst
     * @returns { Float32 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_float32();
        return dst;
    }

    /**
     * @param { Float32 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Float32 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Float32());
    }
}
